spark/Dockerfile

```diff
@@ -50,19 +50,19 @@ WORKDIR ${SPARK_HOME}
 
 ENV SPARK_VERSION=3.5.0
 ENV SPARK_MAJOR_VERSION=3.5
-ENV ICEBERG_VERSION=1.4.3
+ENV ICEBERG_VERSION=1.5.0
 
 # Download spark
 RUN mkdir -p ${SPARK_HOME} \
- && curl https://dlcdn.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.tgz -o spark-${SPARK_VERSION}-bin-hadoop3.tgz \
+ && curl https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.tgz -o spark-${SPARK_VERSION}-bin-hadoop3.tgz \
```

Comment:
Spark 3.5.1 was released, so it's not available anymore at that location, thus we're switching to this URL